packages/@aws-cdk/aws-eks/README.md

```diff
@@ -15,20 +15,173 @@
+explicit `env.region`. See [Environments](https://docs.aws.amazon.com/cdk/latest/guide/environments.html)
+in the AWS CDK Developer Guide for more details.
+
+Here is a [complete sample](./test/integ.eks-kubectl.lit.ts).
+
+### Interacting with Your Cluster
+
+The Amazon EKS construct library allows you to specify an IAM role that will be
+granted `system:masters` privileges on your cluster.
+
+Without specifying a `mastersRole`, you will not be able to interact manually
+with the cluster.
+
+The following example defines an IAM role that can be assumed by all users
+in the account and shows how to use the `mastersRole` property to map this
+role to the Kubernetes `system:masters` group:
+
+```ts
+// first define the role
+const clusterAdmin = new iam.Role(this, 'AdminRole', {
+  assumedBy: new iam.AccountRootPrincipal()
+});
+
+// now define the cluster and make map the role to the "masters" group
+new eks.Cluster(this, 'Cluster', {
+  vpc: vpc,
+  mastersRole: clusterAdmin
+});
+```
+
+Now, given AWS credentials for a user that is trusted by the masters role, you
+should be able to interact with your cluster like this:
+
+```console
+$ aws eks update-kubeconfig --name CLUSTER-NAME
+$ kubectl get all -n kube-system
+...
+```
+
+**NOTE**: if the cluster is configured with `kubectlEnabled: false`, it
+will be created with the role/user that created the AWS CloudFormation
+stack. See [Kubectl Support](#kubectl-support) for details.
+
+### Defining Kubernetes Resources
+
+The `cluster.addManifest()` method can be used to apply Kubernetes resource
+manifests to this cluster.
+
+The following examples will deploy the [paulbouwer/hello-kubernetes](https://github.com/paulbouwer/hello-kubernetes)
+service on the cluster:
+
+```ts
+const appLabel = { app: "hello-kubernetes" };
+
+const deployment = {
+  apiVersion: "apps/v1",
+  kind: "Deployment",
+  metadata: { name: "hello-kubernetes" },
+  spec: {
+    replicas: 3,
+    selector: { matchLabels: appLabel },
+    template: {
+      metadata: { labels: appLabel },
+      spec: {
+        containers: [
+          {
+            name: "hello-kubernetes",
+            image: "paulbouwer/hello-kubernetes:1.5",
+            ports: [ { containerPort: 8080 } ]
+          }
+        ]
+      }
+    }
+  }
+};
+
+const service = {
+  apiVersion: "v1",
+  kind: "Service",
+  metadata: { name: "hello-kubernetes" },
+  spec: {
+    type: "LoadBalancer",
+    ports: [ { port: 80, targetPort: 8080 } ],
+    selector: appLabel
+  }
+};
+
+cluster.addManifest('hello-kub', service, deployment);
+```
+
+Since manifests are modeled as CloudFormation resources. This means that if the
+`addManifest` statement is deleted from your code, the next `cdk deploy` will
+issue a `kubectl delete` command and the Kubernetes resources will be deleted.
+
+You can also use the `KubernetesManifest` construct directly:
+
+```ts
+new KubernetesManifest(this, 'service', {
```

Comment:
How does this associate with a `cluster`? Should there not be a `cluster` parameter?